exchange/bitswap/bitswap.go

```diff
@@ -317,8 +325,11 @@ func (bs *Bitswap) HasBlock(blk blocks.Block) error {
 	// it now as it requires more thought and isnt causing immediate problems.
 	bs.notifications.Publish(blk)
 
-	for _, s := range bs.SessionsForBlock(blk.Cid()) {
-		s.receiveBlockFrom("", blk)
+	k := blk.Cid()
+	ks := []*cid.Cid{k}
+	for _, s := range bs.SessionsForBlock(k) {
+		s.receiveBlockFrom(from, blk)
```

Comment:
hrm... the change from `""` to `from` here is on purpose? Whats the reasoning? ( i dont remember all the implications without my coffee)

Reply:
okay, that sounds good. Any way you could write a test?

Reply:
`TestFetchGraph`? :smile: It's a bit hard to test a race condition...

Reply:
(this is fixing a test failure so there is, by definition, a test case).

Reply:
okay fineeee